itou/siae_evaluations/models.py

```diff
@@ -303,10 +308,26 @@ def __str__(self):
     def review(self):
         if self.state in [evaluation_enums.EvaluatedSiaeState.ACCEPTED, evaluation_enums.EvaluatedSiaeState.REFUSED]:
             with transaction.atomic():
+
+                if self.state == evaluation_enums.EvaluatedSiaeState.ACCEPTED:
+                    if self.reviewed_at:
+                        email = self.get_email_to_siae_reviewed(adversarial=True)
+                    else:
+                        email = self.get_email_to_siae_reviewed(adversarial=False)
+
+                elif self.state == evaluation_enums.EvaluatedSiaeState.REFUSED:
+                    if self.reviewed_at:
+                        email = self.get_email_to_siae_adversarial_stage()
+                    else:
+                        email = self.get_email_to_siae_refused()
```

Comment:
note to self. il doit y avoir une méthode plus pythonic à ces ~if~ imbriqués.

Reply:
tu pourrais mettre un dict :
```
{
    (evaluation_enums.EvaluatedSiaeState.ACCEPTED, True): functools.partial(self.get_email_to_siae_reviewed, adversarial=True),
    (evaluation_enums.EvaluatedSiaeState.ACCEPTED, False): functools.partial(self.get_email_to_siae_reviewed, adversarial=False),
    (evaluation_enums.EvaluatedSiaeState.REFUSED, True): self.get_email_to_siae_adversarial_stage,
    (evaluation_enums.EvaluatedSiaeState.REFUSED, False): self.get_email_to_siae_refused,
}[(self.state, bool(self.reviewed_at))]()
```
ou 
```
{
    evaluation_enums.EvaluatedSiaeState.ACCEPTED: {
        True: functools.partial(self.get_email_to_siae_reviewed, adversarial=True),
        False: functools.partial(self.get_email_to_siae_reviewed, adversarial=False),
    },
    evaluation_enums.EvaluatedSiaeState.REFUSED: {
        True: self.get_email_to_siae_adversarial_stage,
        False: self.get_email_to_siae_refused,
    }
}[self.state][bool(self.reviewed_at)]()
```
Mais bon ici, je ne trouve pas ça vraiment mieux :D

Reply:
Go for fun and to learn something new